src/thor/timedistancematrix.cc

```diff
@@ -387,23 +398,10 @@ void TimeDistanceMatrix::SetDestinations(
   for (const auto& loc : locations) {
     // Set up the destination - consider each possible location edge.
     bool added = false;
-
-    // Only skip outbound edges if we have other options
-    bool has_other_edges = false;
-    std::for_each(loc.correlation().edges().begin(), loc.correlation().edges().end(),
-                  [&has_other_edges](const valhalla::PathEdge& e) {
-                    has_other_edges = has_other_edges || (FORWARD ? !e.begin_node() : !e.end_node());
-                  });
     for (const auto& edge : loc.correlation().edges()) {
-      // If destination is at a node skip any outbound edges
-      if (has_other_edges && (FORWARD ? edge.begin_node() : edge.end_node())) {
-        continue;
-      }
```

Comment:
This was also a bug in the recent templatization.. I kind of forgot there to test TimeDistanceMatrix for trivial cases. And now it showed. We have to include the outbound edges regardless if we have alternatives or not. The `UpdateDestinations` won't be triggered otherwise when source & target is the exact same on line 252. 

FWIW, before templatizing this whole thing there wasn't any check like this on the destinations either. I included it mindlessly without really thinking about it.. Also added a test to see if we're still properly resolving trivial cases, we do:)